include/strf/detail/output_types/FILE.hpp

```diff
@@ -110,8 +110,14 @@ class wide_cfile_writer final: public strf::basic_outbuf_noexcept<wchar_t>
     {
     }
 
-    void recycle() noexcept
+    STRF_HD void recycle() noexcept override
     {
+#ifdef __CUDA_ARCH__
+        asm("trap;");
+#endif
+        // This will only be compiled as device-side code;
+        // the host-side version simply doesn't have object
+        // code, so using it should fail linking
```

Comment:
Couldn't we just not include `output_types/FILE.hpp` in `strf.hpp` for `__CUDA_ARCH__` ?